### ISSUE TITLE
Found unexpected null pointer value while trying to cast value of type 'NSTimer' (0x11c9b2320) to 'Swift.AnyObject' (0x11c984d50)

### ISSUE THREAD

**Issue by spnkr** [johnno1962/InjectionIII#377]
Using 4.1.3, I get a crash every time I edit a Swift UI file. 

```
Found unexpected null pointer value while trying to cast value of type 'NSTimer' (0x11c9b2320) to 'Swift.AnyObject' (0x11c984d50)

2022-02-25 11:52:08.192038-0500 OneDrop[52618:7264637] Found unexpected null pointer value while trying to cast value of type 'NSTimer' (0x11c9b2320) to 'Swift.AnyObject' (0x11c984d50)
```

These are all the references to NSTimer in the workspace:

<img width="1298" alt="image" src="https://user-images.githubusercontent.com/11250/155755909-728ac507-51ed-454a-a86a-99dba26b63c5.png">

Any ideas?

Xcode 13.2.1, in rosetta mode, with a project that worked before. Trigger: I got a new laptop (if this helps).

Full trace:
```
💉 Compiling /Users/.../Something.swift
💉 Loading .dylib ...
💉 Loaded .dylib - Ignore any duplicate class warning ⬆️
💉 Interposed 4 function references.
Found unexpected null pointer value while trying to cast value of type 'NSTimer' (0x11c9b2320) to 'Swift.AnyObject' (0x11c984d50)
2022-02-25 11:52:08.192038-0500 XXX[52618:7264637] Found unexpected null pointer value while trying to cast value of type 'NSTimer' (0x11c9b2320) to 'Swift.AnyObject' (0x11c984d50)
dyld4 config: DYLD_ROOT_PATH=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot DYLD_LIBRARY_PATH=/Users/will/Library/Developer/Xcode/DerivedData/XXX-ebbbrjcrqanmouhfnevadpzbhkbj/Build/Products/Debug-watchsimulator:/Users/will/Library/Developer/Xcode/DerivedData/XXX-ebbbrjcrqanmouhfnevadpzbhkbj/Build/Products/Debug-iphonesimulator:/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/usr/lib/system/introspection DYLD_INSERT_LIBRARIES=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/usr/lib/libBacktraceRecording.dylib:/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/usr/lib/libMainThreadChecker.dylib:/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/usr/lib/libMTLCapture.dylib:/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/Developer/Library/PrivateFrameworks/DTDDISupport.framework/libViewDebuggerSupport.dylib DYLD_FRAMEWORK_PATH=/Users/will/Library/Developer/Xcode/DerivedData/XXX-ebbbrjcrqanmouhfnevadpzbhkbj/Build/Products/Debug-watchsimulator:/Users/will/Library/Developer/Xcode/DerivedData/XXX-ebbbrjcrqanmouhfnevadpzbhkbj/Build/Products/Debug-iphonesimulator:/Users/will/Library/Developer/Xcode/DerivedData/XXX-ebbbrjcrqanmouhfnevadpzbhkbj/Build/Products/Debug-watchsimulator/PackageFrameworks:/Users/will/Library/Developer/Xcode/DerivedData/XXX-ebbbrjcrqanmouhfnevadpzbhkbj/Build/Products/Debug-iphonesim
Found unexpected null pointer value while trying to cast value of type 'NSTimer' (0x11c9b2320) to 'Swift.AnyObject' (0x11c984d50)
CoreSimulator 783.5 - Device: iPhone 13 mini (F24228D5-286F-4106-84B0-C4E789C74083) - Runtime: iOS 15.2 (19C51) - DeviceType: iPhone 13 mini
```

**Comment by johnno1962** [comment]
Hi, the first thing that springs to mind with 4.1.3 relative to previous versions since 3.1.8+ is you have to opt into preserving the values of top level & static vars across injection with 4.1.3. The first thing to try is running your app with a value for the environment variable INJECTION_PRESERVE_STATICS to see if this is the relevant change.

**Comment by spnkr** [comment]
Cool thanks! I'll try that

**Comment by johnno1962** [comment]
What was the version # for the previous version you were using that worked?

**Comment by johnno1962** [comment]
Try the 3.2.1 release. It has the debug files but if would be useful to know if you experienced this problem with that version.

**Comment by spnkr** [comment]
> What was the version # for the previous version you were using that worked?

I'm not sure--I wiped my old laptop. But was def the latest version in Dec or Jan. Excluding the RC releases.

**Comment by spnkr** [comment]
@johnno1962 v3.2.1 works! 💯 

v4.3.1 w/ these env vars still crashes:
<img width="934" alt="image" src="https://user-images.githubusercontent.com/11250/155817308-425f522f-f6fc-46c9-aa8b-ec4f1d4f8504.png">

<img width="832" alt="image" src="https://user-images.githubusercontent.com/11250/155817371-b827ddf7-a1fe-4653-8eb5-6629025a6206.png">

**Comment by spnkr** [comment]
I was looking in the source for `INJECTION_DYNAMIC_CAST`--couldn't find this string. can you point me to the file/area of the repo this is in? maybe I can help figure it out

**Comment by johnno1962** [comment]
D'oh, likes like a plain old regression of some sort then. These opt-in variables were the only designed changes but there has been a lot of water under the bridge since then as I've been getting injection on a device working. If you clone the repo with -`-recurse-submodules` you should find the code related to dynamic cast [here](https://github.com/johnno1962/HotReloading/blob/main/Sources/HotReloading/DynamicCast.swift) which for 4.1.3 shouldn't be in use and for 3.2.1 it was but the strategy I'd recommend is to try to bisect the version where the regression slipped in by trying the precompiled 3.2.2RC*, 4.* releases and I can look at what changed. RC6+ have the .dSYM files until I removed them in 4.1.3 because of the LLDB problem.

**Comment by johnno1962** [comment]
@spnkr, I've rolled two new releases. 3.2.1.1 which is the version you report is working with the very small change required by Xcode 13.3 which will keep you going. I also released 3.2.2 which will be the last of the 3.* series with the Xcode 13.3 fix applied before I got carried away with more sweeping changes to support injection on a device. If you get the time to try the 3.2.2 release if would help narrow down where the problem is.

**Comment by johnno1962** [comment]
On reflection and looking at what's changed this could be the global "sweep" of objects to implement calling the `@objc injected()` method after injection that's failing with 4.1.3. There are two environment more variables INJECTION_SWEEP_DETAIL which prints each object as it is swept and you can use INJECTION_SWEEP_EXCLUDE=. to effectively disable the sweep altogether.
